### DISCUSSION TITLE
Calculated field when saving

### DISCUSSION THREAD

**Discussion by Zyles** [directus/directus#2787] [Archive]
I need to have a field that is calculated based on other fields. By using custom PHP code.

For example I have two fields:
a
b

And the final sum field and the calculation to be done is:
c=a+b

This should happen every time the collection item is updated.

Is this possible?

**Comment by benhaynes** [comment]
Yup! You can process this with a custom hook:

https://docs.directus.io/guides/api-hooks/

**Reply by neo7-studio-web** [reply]
Updated link : https://docs.directus.io/extensions/hooks/
